lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -450,6 +450,49 @@ SPIRVToLLVMDbgTran::transTypeSubrange(const SPIRVExtInst *DebugInst) {
                                      TranslatedOps[2], TranslatedOps[3]);
 }
 
+DIStringType *
+SPIRVToLLVMDbgTran::transTypeString(const SPIRVExtInst *DebugInst) {
+  using namespace SPIRVDebug::Operand::TypeString;
+  const SPIRVWordVec &Ops = DebugInst->getArguments();
+  assert(Ops.size() >= MinOperandCount && "Invalid number of operands");
+
+  StringRef Name = getString(Ops[NameIdx]);
+  DIType *BaseTy =
+      transDebugInst<DIType>(BM->get<SPIRVExtInst>(Ops[BaseTypeIdx]));
+
+  DIExpression *StrLocationExp = nullptr;
+  if (!getDbgInst<SPIRVDebug::DebugInfoNone>(Ops[DataLocationIdx])) {
+    if (const auto *DIExpr =
+            getDbgInst<SPIRVDebug::Expression>(Ops[DataLocationIdx]))
+      StrLocationExp = transDebugInst<DIExpression>(DIExpr);
+  }
+
+  uint64_t SizeInBits = BM->get<SPIRVConstant>(Ops[SizeIdx])->getZExtIntValue();
+
+  DIExpression *StringLengthExp = nullptr;
+  DIVariable *StringLengthVar = nullptr;
+  if (!getDbgInst<SPIRVDebug::DebugInfoNone>(Ops[LengthAddrIdx])) {
+    if (const auto *GV =
+            getDbgInst<SPIRVDebug::GlobalVariable>(Ops[LengthAddrIdx]))
+      StringLengthVar = transDebugInst<DIGlobalVariable>(GV);
+    if (const auto *LV =
+            getDbgInst<SPIRVDebug::LocalVariable>(Ops[LengthAddrIdx]))
+      StringLengthVar = transDebugInst<DILocalVariable>(LV);
+    if (const auto *DIExpr =
+            getDbgInst<SPIRVDebug::Expression>(Ops[LengthAddrIdx]))
+      StringLengthExp = transDebugInst<DIExpression>(DIExpr);
+  }
+
+  if (SizeInBits)
+    return Builder.createStringType(Name, SizeInBits);
+  if (StringLengthVar)
+    return Builder.createStringType(Name, StringLengthVar, StrLocationExp);
+  if (StringLengthExp)
+    return Builder.createStringType(Name, StringLengthExp, StrLocationExp);
```

Comment:
I do believe, if we pass nullptr(s) to createStringType nothing bad will happen. Shouldn't we just call it with every required parameter we have not checking, if it's nullptr?

Reply:
Actually, there are 3 function overloads for `createStringType`, which accept different types and number of arguments.
That follow there are also 3 different representations of `DIStringType` in LLVM IR:
https://llvm.org/doxygen/DIBuilder_8h_source.html#l00213

Reply:
We should return something from the function. We can do it by either creating DIStringType using [DIStringType::get](https://llvm.org/doxygen/classllvm_1_1MDNode.html#a7d10a7b9b7f40b04d27ed97c38ea1950) (and there you can play around adding or not adding elements to vector of metadata), or return nullptr after the assert. Taking in account Encoding I'd suggest to take a look at 'get' function approach.